editions/tw5.com/tiddlers/macros/UnusedTitleMacro.tid

```diff
@@ -10,7 +11,10 @@ It uses the same method as the create new tiddler button, a number is appended t
 
 !! Parameters
 
-;baseName
+; baseName
 : A string specifying the desired base name, defaulting to `New Tiddler`
 
+; prefix
+: A string specifying the separator between baseName and the unique number. eg: `prefix:"-"`. The macro is ''does not allow'' an ''empty'' string!  <<.from-version "5.1.24">>
```

Comment:
The macro does allow an empty string, but `.wiki.generateNewTitle()` replaces it with a `space`. I didn't change this behaviour because of backwards compatibility.

Reply:
It does look as though your code now does do the right thing with an empty prefix passed to generateNewTitle, is that not the case?

Reply:
Ah I see what you mean, you're talking about `separator = separator || " ";` in the macro. It seems like it would be better to test for undefined (I'm pretty sure macros get undefined for missing parameters).

Reply:
If `options.prefix` is an empty string the .wiki.js function will overwrite it. 

see: https://github.com/Jermolene/TiddlyWiki5/blob/e84c87ef37e63473e111a37292ceb977eeb87bd2/core/modules/wiki.js#L200 

If we would change this behaviour, it would break backwards compatibility for everyone that uses. `$tw.wiki.generateNewTitle()` in plugins or other code. That's the reason why I did add some info to the docs. So users know, that it's not a bug.

Reply:
>(I'm pretty sure macros get undefined for missing parameters).

Nope, macros get an empty string. I think that was 1 reason why I couldn't implement an empty string